packages/react/src/exchanges/hooks/useCreateExchangeFilter.ts

```diff
@@ -36,27 +51,30 @@ function useCreateExchangeFilter() {
         return Promise.reject(new Error('No data to filter provided'));
       }
 
-      reset();
+      if (!orderItemUuid) {
+        return Promise.reject(new Error('No orderItemUuid provided'));
+      }
 
-      return createExchangeFilter(exchangeFilterData, config);
+      return createExchangeFilter(exchangeFilterData, orderItemUuid, config);
     },
-    [createExchangeFilter, reset],
+    [createExchangeFilter, orderItemUuid],
   );
 
   const data = useMemo(() => {
-    if (!exchangeFilterResult) {
+    if (!exchangeFilters && !exchangeFilterById) {
       return undefined;
     }
 
-    return exchangeFilterResult;
-  }, [exchangeFilterResult]);
+    return { exchangeFilters, exchangeFilterById };
```

Comment:
Why the `useCreateExchangeFilter` needs to return all exchange filters? This should return only the new exchange filter. If the user needs to have all exchange filters, then we should have a hook called `useExchangeFilters` and one of its actions should be `create`.

Reply:
We talked in private and decided to rename this hook and change its behaviour. It returns all exchange filters only.